unit/src/DependencyResolverTest.C

```diff
@@ -144,3 +144,21 @@ TEST_F(DependencyResolverTest, dependsOnTest)
   EXPECT_FALSE(_tree.dependsOn("m1", "something_else"));
   EXPECT_FALSE(_tree.dependsOn("something_else", "k0"));
 }
+
+TEST_F(DependencyResolverTest, cyclicTest)
+{
+  try
+  {
+    // Attempt to insert a dependency that results in cyclicity
+    _resolver.insertDependency("a", "b");
+    FAIL() << "missing expected exception";
+  }
+  catch (const std::exception & e)
```

Comment:
So are you saying that i should throw CyclicDependencyException instead of just doing mooseError ?   (It's easier to ask rather than just doing it because any change to DependencyResolver.h takes basically a day to compile, so i have to be careful when changing these "core" header files.)